Smoothing code.cpp

```diff
@@ -31,6 +31,31 @@ struct MovingAverage {
 std::uint16_t get_reading() {
     // this always returns 7 just because.... put your "read the pin" code here
     return 7;
+  int sensorPin = A0;
+int ledPin = LED_BUILTIN;
+int sensorValue = 0;
+void setup() {
+  // put your setup code here, to run once:
+  Serial.begin(9600);
+  pinMode (sensorPin, INPUT);
+  pinMode(ledPin,OUTPUT);
+}
+
+void loop() {
```

Comment:
same as with `setup()`, this is a function inside a function which is invalid.
BUT! the code _inside_ this function (`analogRead`) does go into `get_reading()`.
Basically, the `get_reading()` call just does the `analogRead` and returns what it read 

it was my mistake for not renaming `main()` in my example to `loop()`. I originally wrote it that way, but I wanted to run the code before I sent it :) normal programs use `main()` as their entry to the program but arduino is different